64_FFT/app_resources/shader.comp.hlsl

```diff
@@ -39,41 +39,13 @@ struct Accessor {
 void main(uint32_t3 ID : SV_DispatchThreadID)
 {
 
-	// Workgroup
-
-	if (nbl::hlsl::workgroup::SubgroupContiguousIndex() >= pushConstants.dataElementCount)
-		return;
-
-	// Subgroup (works fine)
-	
-	/*
-	if (nbl::hlsl::glsl::gl_SubgroupInvocationID() >= pushConstants.dataElementCount)
-		return;
-	*/
-
-	// Workgroup	
-
 	Accessor accessor;
 	SharedMemoryAccessor sharedmemAccessor;
-	nbl::hlsl::workgroup::FFT<2, false, input_t>::template __call<Accessor, SharedMemoryAccessor>(accessor, sharedmemAccessor);	
-	//nbl::hlsl::workgroup::FFT<2, true, input_t>::template __call<Accessor, SharedMemoryAccessor>(accessor, sharedmemAccessor);
-	
-	// Subgroup (works fine)
-	/*
 
-	nbl::hlsl::complex_t<input_t> lo, hi;
-	lo.real(accessor.get(nbl::hlsl::glsl::gl_SubgroupInvocationID()));
-	hi.real(accessor.get(nbl::hlsl::glsl::gl_SubgroupInvocationID() + _NBL_HLSL_WORKGROUP_SIZE_));
-	lo.imag(accessor.get(nbl::hlsl::glsl::gl_SubgroupInvocationID() + 2 * _NBL_HLSL_WORKGROUP_SIZE_));
-	hi.imag(accessor.get(nbl::hlsl::glsl::gl_SubgroupInvocationID() + 3 * _NBL_HLSL_WORKGROUP_SIZE_));
-	
-	nbl::hlsl::subgroup::FFT<false, input_t>::__call(lo, hi);
-	//nbl::hlsl::subgroup::FFT<true, input_t>::__call(lo, hi);
+	// Workgroup	
 
-	accessor.set(nbl::hlsl::glsl::gl_SubgroupInvocationID(), lo.real());
-	accessor.set(nbl::hlsl::glsl::gl_SubgroupInvocationID() + _NBL_HLSL_WORKGROUP_SIZE_, hi.real());
-	accessor.set(nbl::hlsl::glsl::gl_SubgroupInvocationID() + 2 * _NBL_HLSL_WORKGROUP_SIZE_, lo.imag());
-	accessor.set(nbl::hlsl::glsl::gl_SubgroupInvocationID() + 3 * _NBL_HLSL_WORKGROUP_SIZE_, hi.imag());		
+	nbl::hlsl::workgroup::FFT<2, true, input_t>::template __call<Accessor, SharedMemoryAccessor>(accessor, sharedmemAccessor);
+	accessor.workgroupExecutionAndMemoryBarrier();
+	nbl::hlsl::workgroup::FFT<2, false, input_t>::template __call<Accessor, SharedMemoryAccessor>(accessor, sharedmemAccessor);	
```

Comment:
you still haven't swapped the input and output buffers

which means the source for your inverse FFT is not the output of the forward FFT

Reply:
@Fletterio did you fix this?

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla-Examples-and-Tests/pull/118#discussion_r1654728895